src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -468,16 +469,21 @@ public override int Read(char[] buffer, int index, int count)
                     throw new ArgumentNullException(nameof(buffer));
                 }
 
-                if (index < 0 || index >= buffer.Length)
```

Comment:
Note: I changed this because it was wrong. I should be able to read into a zero-length array when `count` is 0. Or when `index` is equal to `buffer.Length` and `count` is 0.

Also, by .NET convention, if `index + count` is greater than `buffer.Length`, the exception that is thrown shouldn't have `index` or `count` as its parameter. I should be an `ArgumentException`  with a special message without a parameter (because it related to both parameters, or rather all 3 and the relationship between them).

See: https://source.dot.net/#System.Private.CoreLib/StringReader.cs,86

Reply:
thanks for the great explanation :)